// ndn-repo/Gruntfile.js
module.exports = function(grunt){


   var browsers = [{
        browserName: "chrome",
        version: "33",
        platform: "XP"
    }, {
      browserName: "chrome",
      version: "33",
      platform: "Linux"
    }];

  grunt.initConfig({
    browserify: {
      suite: {
        src: "test/node/*.js"
        , dest: "test/browser/suite.js"
      }
    },
    jsdoc : {
      dist : {
        src: ['src/**/*.js'],
        options: {
          destination: 'doc'
        }
      }
    },
    mochaTest: {
      suite: {
        options: {
          reporter: 'spec'
          ,clearRequireCache: true
        },
        src: ["test/node/*.js"]
      },
    },
    watch: {
      NameTree: {
        files: ['src/**/*.js', 'src/*.js'],
        tasks: ['jshint', 'browserify','mochaTest']
      },
      ContentStore: {
        files: ['src/DataStructures/ContentStore.js', 'test/node/ContentStore.js'],
        tasks: ['jshint:ContentStore', 'browserify:testDataStructures','mochaTest:DataStructures']
      },
      FIB: {
        files: ['src/DataStructures/FIB.js', 'test/node/FIB.js'],
        tasks: ['jshint:FIB', 'browserify:testDataStructures','mochaTest:DataStructures']
      },
      PIT: {
        files: ['src/DataStructures/PIT.js', 'test/node/PIT.js'],
        tasks: ['jshint:PIT', 'browserify:testDataStructures','mochaTest:DataStructures']
      },
      Interfaces: {
        files: ['src/DataStructures/Interfaces.js', 'test/node/Interfaces.js'],
        tasks: ['jshint:Interfaces', 'browserify:testDataStructures','mochaTest:DataStructures']
      },
      Transports: {
        files: ["src/Transports/*.js", 'src/Transports/*.js'],
        tasks: ['jshint:Transports', 'browserify:testTransports' ]
      },
      livereload: {
        // Here we watch the files the sass task will compile to
        // These files are sent to the live reload server after sass compiles to them
        options: { livereload: true },
        files: ['test/browser/*.js'],
      },
    },
    jshint: {
      options: {
        curly: true,
        eqeqeq: true,
        laxcomma: true,
        laxbreak: true
      },
      NameTree: ['src/*.js','src/**/*.js']
    }


  })
  grunt.loadNpmTasks('grunt-jsdoc');
  grunt.loadNpmTasks("grunt-mocha-test");
  grunt.loadNpmTasks('grunt-contrib-jshint');
  grunt.loadNpmTasks("grunt-contrib-watch");

  grunt.loadNpmTasks('grunt-browserify');

  grunt.registerTask("brow", ["browserify:test"])
  grunt.registerTask("test", "mochaTest");
  grunt.registerTask("hint", "jshint")
  grunt.registerTask("suite", ["brow", "test"])
}
